src/experimentExtractor/index.js

```diff
@@ -0,0 +1,128 @@
+const fs = require("fs-extra");
+const {
+  EXPERIMENT_COMPONENT_NAME,
+  EXPERIMENTS_FILE_NAME,
+} = require("./constants");
+const { extractExperimentData } = require("./parser");
+
+// TODO: move to utils and add tests
+const chunkNameToPageNameRegex = (chunkName) => {  
+  // replacing all \\ to a /
+  let normalizedChunkName = chunkName.replace(/\\+/gi, "/");
+  // remove .js extension 
+  normalizedChunkName = normalizedChunkName.replace(/\.\S+$/gi, "");
+  // replace 'pages' prefix with ^
+  normalizedChunkName = normalizedChunkName.replace(/^pages/gi, "^");
+  // replace all / with \/
+  normalizedChunkName = normalizedChunkName.replace(/\//gi, "\\/");
+  // replace [__anything__] with [^\s\/]+
+  normalizedChunkName = normalizedChunkName.replace(/\[\S+\]/gi, "[^\\s\\/]+");
+  // add $ at the end
+  normalizedChunkName = `${normalizedChunkName}$`
+  return normalizedChunkName;
+};
+
+const PLUGIN_NAME = "ExperimentExtractorPlugin";
+const LIBRARY_NAME = "next-experiments";
+
+class ExperimentExtractorPlugin {
+  constructor(
+    options = { resultFilePath: EXPERIMENTS_FILE_NAME, isDebug: false, libraryName: LIBRARY_NAME }
+  ) {
+    this.resultFilePath = options.resultFilePath;
+    this.isDebug = options.isDebug;
+    this.libraryName = options.libraryName;
```

Comment:
Is this just `this = pick(options, ['resultFilePath', 'isDebug', 'libraryName']`?

Reply:
As far as I know, we can't assign directly to `this`.
TS compiler just confirmed that.